### DISCUSSION TITLE
Colour contrast map areas that have been downloaded or not

### DISCUSSION THREAD

**Discussion by tawmjk** [organicmaps/organicmaps#4360] [Ideas]
When I am zoomed out to see most of a country, I cannot tell if I have installed a map for a region.  I'd like to see some highlighting of any areas that I have downloaded.  For example, if I had downloaded maps for Frankfurt and nearby areas, when I zoomed out enough to see Cologne too then I could tell that this area map had not been downloaded.

**Comment by biodranik** [comment]
Please search for existing issues, I remember we already had this one open.

**Reply by tawmjk** [reply]
Found it, I thought this was an idea not an issue.  I did not see any ideas like this so made a new suggestion.

**Comment by oleg-rswll** [comment]
> When I am zoomed out to see most of a country, I cannot tell if I have installed a map for a region. I'd like to see some highlighting of any areas that I have downloaded. For example, if I had downloaded maps for Frankfurt and nearby areas, when I zoomed out enough to see Cologne too then I could tell that this area map had not been downloaded.

Can you share the goal for seeing the downloaded and un-downloaded maps, is it to easier discover what is not downloaded so you can download?  In what scenarios or what flow are you working on when this becomes needed?

**Reply by biodranik** [reply]
There are several important use cases:
1. Highlight outdated maps so users will be motivated to update. Old maps data often leads to wrong routes, bad search results, does not have fresh OSM fixes, etc.
2. Highlight downloaded areas. As our router uses only downloaded map data, it will build a longer route if the map between two points is not downloaded, but surrounding maps are downloaded.

**Comment by tawmjk** [comment]
An example would be travelling to a country with over 40 maps and I don't want to load them all just the few in the areas that I want to visit and other maps between two or more locations that I want to visit.

The easiest answer is to navigate a route and then the relevant maps are downloaded but even then there might be a few more that I need and don't know have been downloaded.   The other is to zoom in to see the detailed maps and scroll around but that might mean a map is missed.

**Reply by oleg-rswll** [reply]
Thank you, that's helpful.

**Comment by oleg-rswll** [comment]
This type of view is quite necessary.  This likely should not be the default view, as most maps would be missing and it may not be a good look. This can be addressed by adding a new layer that shows the downloaded maps in a certain color shade, outdated maps in another color and the non-downloaded maps in a greyed out shade.  And it would display this way no matter how zoomed in or zoomed out the map is.

**Reply by tawmjk** [reply]
I like it!

**Comment by pastk** [comment]
- #1356
